common/src/main/scala/org/apache/celeborn/common/quota/ResourceConsumption.scala

```diff
@@ -17,26 +17,67 @@
 
 package org.apache.celeborn.common.quota
 
-import org.apache.celeborn.common.util.Utils
+import java.util
+
+import scala.collection.JavaConverters._
+
+import org.apache.celeborn.common.util.{CollectionUtils, Utils}
 
 case class ResourceConsumption(
     diskBytesWritten: Long,
     diskFileCount: Long,
     hdfsBytesWritten: Long,
-    hdfsFileCount: Long) {
+    hdfsFileCount: Long,
+    subResourceConsumptions: util.Map[String, ResourceConsumption] = null) {
 
   def add(other: ResourceConsumption): ResourceConsumption = {
     ResourceConsumption(
       diskBytesWritten + other.diskBytesWritten,
       diskFileCount + other.diskFileCount,
       hdfsBytesWritten + other.hdfsBytesWritten,
-      hdfsFileCount + other.hdfsFileCount)
+      hdfsFileCount + other.hdfsFileCount,
+      subResourceConsumptions(other))
+  }
+
+  private def subResourceConsumptions(other: ResourceConsumption)
+      : util.Map[String, ResourceConsumption] = {
+    var resourceConsumptions: util.Map[String, ResourceConsumption] = null
+    if (subResourceConsumptions == null) {
+      resourceConsumptions = other.subResourceConsumptions
+    } else {
+      resourceConsumptions = subResourceConsumptions
```

Comment:
If use subResourceConsumptions directly， IMO it may lead to repeated resource consumption calculations.